### PR TITLE
Added Progress element docs

### LINKED ISSUE #1056: Add documentation for the progress control
**Description**

[This issue](https://github.com/Avaiga/taipy/issues/692) introduced the new `progress` control.
We need to document it.

**Expected change**

The documentation has a page dedicated to the `progress` control, including different use cases.
That includes references to code examples introduced in [this issue](https://github.com/Avaiga/taipy/issues/1464).

### LINKED ISSUE #1464: Docs for "Progress" visual element are missing
### Issue Description

The docs for a newly created Progress element are missing, here's the PR for the vis element - https://github.com/Avaiga/taipy/pull/1303

### Screenshots or Examples (if applicable)

_No response_

### Proposed Solution (optional)

On taipy.io, we can add a the docs for progress element.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by yaten2302**
# Fixes https://github.com/Avaiga/taipy/issues/1464 
# Fixes https://github.com/Avaiga/taipy-doc/issues/1056

### CODE REVIEW

File: docs/manuals/userman/gui/viselements/generic/progress.md_template [hunk 1]
```diff
@@ -12,4 +12,22 @@ name to select the date selectors on your page and apply style.
 
 # Usage
 
-TODO
+## Circular Progress
```
Comment (FabienLelaquais) [inline docs/manuals/userman/gui/viselements/generic/progress.md_template:15]: Ok so typically, every element has its first use case addressed as 'Simple'.
We try to capture how the element will be mostly used. The best is to have as few properties as possible.
In your situation, that would be: only *value* is set.

Note that *render* should not be mentioned: it is used in very rare occasions to hide an element from a page. Not the topic of this documentation page.

File: docs/manuals/userman/gui/viselements/generic/progress.md_template [hunk 2]
```diff
@@ -12,4 +12,22 @@ name to select the date selectors on your page and apply style.
...
+
+The progress render, which is set by default to True, simply handles that if the progress element should be displayed or not.
+
+!!! taipy-element
```
Comment (FabienLelaquais) [inline docs/manuals/userman/gui/viselements/generic/progress.md_template:17]: This is useless: *render* is not really a feature of the progress bar, and there's no *value* here, which is mandatory.

File: docs/manuals/userman/gui/viselements/generic/progress.md_template [hunk 3]
```diff
@@ -12,4 +12,22 @@ name to select the date selectors on your page and apply style.
...
+    render=True
+
+!!! taipy-element
+    value=50
+    render=True
```
Comment (FabienLelaquais) [inline docs/manuals/userman/gui/viselements/generic/progress.md_template]:
```suggestion
```

File: docs/manuals/userman/gui/viselements/generic/progress.md_template [hunk 4]
```diff
@@ -12,4 +12,22 @@ name to select the date selectors on your page and apply style.
...
+
+## Linear Progress
+
+You can specify if you want the progress element to be linear or circular just by adding - `linear=True`. By default linear is set to False.
+
+!!! taipy-element
+    render=True
```
Comment (FabienLelaquais) [inline docs/manuals/userman/gui/viselements/generic/progress.md_template]:
```suggestion
```

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Ok so that's a start.
The point is to create tine example that shows how a given property (or group of related properties) can be used, with what effect.

The progress control is easy enough: only *value*, *linear* and *show_value* have to have their tiny demo.

**Comment by yaten2302** [comment]
@FabienLelaquais , made the changes 👍 kindly review

File: docs/manuals/userman/gui/viselements/generic/progress.md_template [hunk 5]
```diff
@@ -17,17 +17,14 @@ name to select the date selectors on your page and apply style.
 The progress render, which is set by default to True, simply handles that if the progress element should be displayed or not.
 
 !!! taipy-element
```
Comment (FabienLelaquais) [inline docs/manuals/userman/gui/viselements/generic/progress.md_template:17]: So this is useless right?

**Review by FabienLelaquais — APPROVED** [review]
👍

File: docs/manuals/userman/gui/viselements/generic/progress.md_template (hunk 5 above)
Comment (yaten2302) [reply docs/manuals/userman/gui/viselements/generic/progress.md_template:17]: Made the changes @FabienLelaquais 👍

**Comment by yaten2302** [comment]
Hey @FabienLelaquais , ig I forgot to add the screenshots in here!! Like, should I create a sample app in taipy and use the `progress` element and then take the ss of the rendered element, right?

**Comment by FabienLelaquais** [comment]
> use

Don't bother @yaten2302.
I'm the one that usually build these images for users to rapidly evaluate the effects of setting properties.
I spend quite some time trying to come up with very simple example focusing on a tiny set of features.
And I'm very meticulous on this topic - so whatever you provide, I'll certainly have to redo the job... because that's what I feel.

So please do not do anything. I will take care of this. Thanks!

**Comment by yaten2302** [comment]
Thanks for merging the PR @FabienLelaquais 👍

**Comment by FabienLelaquais** [comment]
Thanks to you, @yaten2302
There is still some work on this (layout and label management) that we will handle internally because it impacts areas of the implementation such as styling that needs discussions...
So do expect this control to grow a little bit in the next few weeks.

**Comment by yaten2302** [comment]
Sure @FabienLelaquais 👍

**Pull request merged**